src/nvim/screen.c

```diff
@@ -6887,6 +6888,10 @@ static void draw_tabline(void)
   }
   redraw_tabline = false;
 
+  if (ui_is_external(kUITabline)) {
+    ui_ext_tabline_update();
+    return;
+  }
```

Comment:
@dzhou121 I moved this below `redraw_tabline = false` to avoid unnecessary `draw_tabline()` calls. Did you have it swapped on purpose?